tests/model_fields/models.py

```diff
@@ -469,3 +471,121 @@ class UUIDChild(PrimaryKeyUUIDModel):
 
 class UUIDGrandchild(UUIDChild):
     pass
+
+
+class GeneratedModel(models.Model):
+    a = models.IntegerField()
+    b = models.IntegerField()
+    field = models.GeneratedField(expression=F("a") + F("b"), db_persist=True)
+
+    class Meta:
+        required_db_features = {
+            "supports_generated_columns",
+            "supports_stored_generated_columns",
+        }
+
+
+class GeneratedModelParams(models.Model):
+    field = models.GeneratedField(
+        expression=Value("Constant", output_field=CharField(max_length=10))
+    )
+
+    class Meta:
+        required_db_features = {
+            "supports_generated_columns",
+            "supports_generated_columns_params",
+            "supports_stored_generated_columns",
+        }
+
+
+class GeneratedModelVirtual(models.Model):
+    name = models.CharField(max_length=10)
+    lower_name = models.GeneratedField(expression=Lower("name"), db_persist=False)
```

Comment:
I removed the `output_field=TextField()` here since it was breaking Oracle and didn't seem required for the tests using it.

```
ORA-54003: specified data type is not supported for a virtual column
```